packages/hardhat-core/src/utils/source-names.ts

```diff
@@ -169,6 +170,19 @@ export function isAbsolutePathSourceName(sourceName: string): boolean {
   return path.isAbsolute(sourceName) || sourceName.startsWith("/");
 }
 
+/**
+ * This function returns true if the sourceName contains the current package's name
+ * as a substring
+ */
+export async function includesOwnPackageName(sourceName: string): Promise<boolean> {
+  const packageName = await getPackageName(sourceName);
+  if (packageName) {
+    return sourceName.includes(packageName);
```

Comment:
Shouldn't this be includes `${packageName}/`?

Reply:
Maybe ```startsWith(`${packageName}/`)``` would be a bit safer.

Reply:
good point! I will update the PR over the weekend